Source/DataSources/KmlDataSource.js

```diff
@@ -1474,27 +1635,298 @@ define([
         console.log('KML - Unsupported feature: ' + node.localName);
     }
 
+    var RefreshMode = {
+        INTERVAL : 0,
+        EXPIRE : 1,
+        STOP : 2
+    };
+
+    var scratchCartographic = new Cartographic();
+    function computeViewRectangle(camera) {
+        if (!defined(camera)) {
+            return Rectangle.MAX_VALUE;
+        }
+
+        var region = camera.computeViewRegion();
+        if (!defined(region)) {
+            return Rectangle.MAX_VALUE;
+        }
+
+        var result;
+        var ellipsoid = Ellipsoid.WGS84;
+        var count = region.length;
+        for(var i=0;i<count;++i) {
+            ellipsoid.cartesianToCartographic(region[i], scratchCartographic);
+            if (!defined(result)) {
+                result = new Rectangle(scratchCartographic.longitude, scratchCartographic.latitude,
+                    scratchCartographic.longitude, scratchCartographic.latitude);
+            } else {
+                Rectangle.expand(result, scratchCartographic, result);
+            }
+        }
+
+        return result;
+    }
+
+    function makeQueryString(string1, string2, addQuestionMark) {
+        function cleanupString(s) {
+            if (!defined(s) || s.length === 0) {
+                return '';
+            }
+
+            var sFirst = s.charAt(0);
+            if (sFirst === '&') {
+                s.splice(0, 1);
```

Comment:
I don't think `splice` is part of the JS spec for strings, use `substr` or `substring` instead.